File: Lesson_5/Class_work/Task_1/CalculatorView.java
package Lesson_5.Class_work.Task_1;

import java.text.DecimalFormat;
import java.util.Scanner;

public class CalculatorView {

    Scanner scan = new Scanner(System.in);


//    public String getOperate() {
//        System.out.print("Enter an operator (+, -, *, /): ");
//        return scan.nextLine();
//    }

    public char getInputOperator() {
        System.out.print("Enter an operator (+, -, *, /): ");
        char c = scan.next().charAt(0);
        switch (c) {
            case '+':
                return c;
            case '*':
                return c;
            case '/':
                return c;
            case '-':
                return c;
            default:
                System.out.println("Invalid operator, try again");
                return getInputOperator();
        }
    }

    public void displayResult(double result) {
        System.out.println("Result: " + result);
    }

    void except() {
        System.out.println("Введено не число\nвведите число");

    }

    public double getDouble() {

        System.out.print("Enter a number 1: ");


        try {
            String n = scan.nextLine();
            double m = Double.parseDouble(n);
            return m;
        } catch (Exception err) {

            except();
            return getDouble();
        }
    }

    public void ytr(double x) {
        System.out.println(x);
    }

    public double getDouble_2() {

        System.out.print("Enter a number 2:");

        try {
            String n = scan.nextLine();
            double m = Double.parseDouble(n);
            return m;
        } catch (Exception err) {
            except();
            return getDouble_2();
        }
    }
}

File: Lesson_7/Task_3/CalculatorView.py
class CalculatorView ():




# //    public String getOperate() {
# //        System.out.print("Enter an operator (+, -, *, /): ");
# //        return scan.nextLine();
# //    }


    def getInputOperator(self):
        print("Enter an operator (+, -, *, /): ");
        c = input()
        if c=="+"or c=="-" or c=="/" or c=="*":
            return c
        else:
            print("Invalid operator, try again");
            return self.getInputOperator()



    def displayResult(self,result) :
        print("Result: " + result);


    def error_text(self) :
        print("Введено не число\nвведите число");



    def  getDouble(self) :

        print("Enter a number 1: ");


        try:
            n=input()
            return n
        except :

            Exception
            self.error_text()
            return self.getDouble()
